### PR TITLE
Add the `apply_pil` method

### PR DESCRIPTION
**Pull request by mraspaud**
This new method delays the processing with a dask delayed object.

 - [x] Tests added (for all bug fixes or enhancements)
 - [x] Tests passed (for all non-documentation changes)
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff`` (remove if you did not edit any Python files)
 - [x] Fully documented (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

### CODE REVIEW

File: trollimage/xrimage.py [hunk 1]
```diff
@@ -417,6 +418,43 @@ def pil_save(self, filename, fformat=None, fill_value=None,
...
+        return np.array(new_img) / self.data.dtype.type(255.0)
+
+    def apply_pil(self, fun, output_mode, pil_args=None, pil_kwargs=None, fun_args=None, fun_kwargs=None):
+        """Apply a function `fun` on the pillow image corresponding to the instance of the XRImage.
+
+        The function shall take a pil image as first argument, and is then passed fun_args and fun_kwargs.
+        It is expected to return the modified pil image.
+        This function returns a new XRImage instance with the modified image data.
+
+        The pil_args and pil_kwargs are passed the the `pil_image` method the XRImage instance.
```
Comment (djhoese) [inline trollimage/xrimage.py]: `the the` -> `to the`

Comment (djhoese) [reply trollimage/xrimage.py]: `of the XRImage instance`

**Comment by mraspaud** [comment]
2. The attributes are included, I think that's the easiest for the user. The user can always modify them afterwards

File: trollimage/xrimage.py (hunk 1 above)
Comment (mraspaud) [reply trollimage/xrimage.py]: Thanks, fixed

**Comment by mraspaud** [comment]
1. The XRImage constructor makes sure the 'bands' dim is always present...

**Comment by djhoese** [comment]
For 1, I was thinking things like "pressure_level" or "time" or other dimensions beyond the y, x, bands dimensions. I suppose these aren't really *images* though, right?

For 2, what I meant was should the `attrs` be passed to the user's function. It doesn't look like they are now, right?

**Comment by mraspaud** [comment]
1. Ah ok, no, I think that shouldn't be handled at the trollimage level, which is just made for images
2. Good idea, I can add that

File: trollimage/xrimage.py [hunk 2]
```diff
@@ -428,7 +428,9 @@ def _delayed_apply_pil(self, fun, pil_args, pil_kwargs, fun_args, fun_kwargs, ou
             fun_args = tuple()
         if fun_kwargs is None:
             fun_kwargs = dict()
-        new_img = fun(self.pil_image(*pil_args, **pil_kwargs), *fun_args, **fun_kwargs)
+        if image_mda is None:
+            image_mda = dict()
+        new_img = fun(self.pil_image(*pil_args, **pil_kwargs), *fun_args, image_mda=image_mda, **fun_kwargs)
```
Comment (djhoese) [inline trollimage/xrimage.py]: Is there a benefit to making this a keyword argument? I envisioned it being the argument right after the `Image` object.

`image_mda` versus `image_metadata`?

Comment (mraspaud) [reply trollimage/xrimage.py]: No benefit

Comment (mraspaud) [reply trollimage/xrimage.py]: fixed

**Pull request merged**